katsdptelstate/telescope_state.py

```diff
@@ -251,6 +273,15 @@ def decode_value(value, allow_pickle=True):
         except Exception as error:
             raise DecodeError(str(error))
     elif value[:1] <= ENCODING_PICKLE:
+        if allow_pickle is None:
+            allow_pickle = ALLOW_PICKLE
+            if WARN_ON_PICKLE:
+                warnings.warn(
+                    'The telescope state contains pickled values. This is a security risk, '
+                    'but is allowed because MeerKAT data up to March 2019 uses it. '
+                    'You can suppress this warning by setting KATSDPTELSTATE_ALLOW_PICKLE=1 '
+                    'in the environment, or disable pickles by setting '
+                    'KATSDPTELSTATE_ALLOW_PICKLE=0.')
```

Comment:
I would disable WARN_ON_PICKLE after the first warning, otherwise every metadata action on katdal will spit out this warning. At least I'm pleasantly surprised that each action only produces a single warning (thanks, `warnings` module!).

Sure, one can disable it via the environment variable, but are you going to do that every time you log into a new machine / container to check something? Instead of passing the burden of an admittedly troublesome design decision on to the end user, I would like to see katdal produce this warning once at the first encounter of a pickle and then hold its peace. This will achieve its goal with the least amount of irritation.

Reply:
The warnings module takes care of that - it'll only warn the first time the warning is encountered.

Reply:
Oh, I see the default for the warnings module is "print the first occurrence of matching warnings for each location where the warning is issued". Are you finding that you're getting multiple warnings because each use from katdal is considered a different location?

Reply:
I think so, yes. Witness the IPython 7.2 + Python 3.7 output below:

```
In [5]: f = katdal.open('1543471660_sdp_l0.full.rdb', chunk_store=None)                               
/Users/schwardt/git/katsdptelstate/katsdptelstate/telescope_state.py:280: UserWarning: The telescope state contains pickled values. This is a security risk, but is allowed because MeerKAT data up to March 2019 uses it. You can suppress this warning by setting KATSDPTELSTATE_ALLOW_PICKLE=1 in the environment, or disable pickles by setting KATSDPTELSTATE_ALLOW_PICKLE=0.
  'The telescope state contains pickled values. This is a security risk, '

In [6]: f.az                                                                                          
/Users/schwardt/git/katsdptelstate/katsdptelstate/telescope_state.py:280: UserWarning: The telescope state contains pickled values. This is a security risk, but is allowed because MeerKAT data up to March 2019 uses it. You can suppress this warning by setting KATSDPTELSTATE_ALLOW_PICKLE=1 in the environment, or disable pickles by setting KATSDPTELSTATE_ALLOW_PICKLE=0.
  'The telescope state contains pickled values. This is a security risk, '
```
The initial string `'The telescope state contains pickled values. This is a security risk, '` also appears twice (maybe it is taken as the 'title' of the warning somehow?).

Reply:
Ok I'll look into it. What you see the second time is the line at which the warning was generated. I should maybe introduce a constant for the warning text so that you don't see part of the message twice.

Reply:
This looks better now, thanks!